### PR TITLE
cli shell using typer

### PR DESCRIPTION
**Pull request by Juliamg**
## Description
Adding shell for cli using typer

### CODE REVIEW

File: cognite/powerops/main.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@app.command(
+    "plan",
+    help="Compare CDF resource creation between desired state (configuration files)"
+    " and remote state (resources in CDF) and print differences",
+)
+def plan(
+    path: Annotated[Path, typer.Argument(help="Path to configuration files")],
+):
+    typer.echo(f"Running plan on configuration files located at {path}")
```
Comment (haukb) [inline cognite/powerops/main.py:17]:
```suggestion
    typer.echo(f"Running plan on configuration files located in {path}")
```

File: cognite/powerops/main.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+@app.command("apply", help="Apply differences between desired and actual state. Writes desired state to CDF")
+def apply(path: Annotated[Path, typer.Argument(help="Path to configuration files")]):
+    typer.echo(f"Running apply on configuration files located at {path}")
```
Comment (haukb) [inline cognite/powerops/main.py:22]:
```suggestion
    typer.echo(f"Running apply on configuration files located in {path}")
```

File: cognite/powerops/main.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    help="Compare CDF resource creation between desired state (configuration files)"
+    " and remote state (resources in CDF) and print differences",
+)
+def plan(
+    path: Annotated[Path, typer.Argument(help="Path to configuration files")],
+):
+    typer.echo(f"Running plan on configuration files located at {path}")
+
+
+@app.command("apply", help="Apply differences between desired and actual state. Writes desired state to CDF")
```
Comment (haukb) [inline cognite/powerops/main.py:20]: Add parenthesis here as well. Just to be very explicit what is desired and actual state. 
```suggestion
@app.command("apply", help="Apply the changes from the configuration files to the data model in CDF")
```

File: cognite/powerops/main.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import typer
+
+app = typer.Typer()
+
+
+@app.command(
+    "plan",
+    help="Compare CDF resource creation between desired state (configuration files)"
+    " and remote state (resources in CDF) and print differences",
```
Comment (haukb) [inline cognite/powerops/main.py:12]: Just a suggestion to the wording. Feel free to decline or rewrite again.
```suggestion
    help="Show the effect of applying the changes from the configuration files to the data model in CDF",
```

**Review by haukb — CHANGES_REQUESTED** [review]

**Review by doctrino — CHANGES_REQUESTED** [review]
Update changelog and version

**Review by haukb — APPROVED** [review]
perfect!

**Pull request merged**